Code/GraphMol/SynthonSpaceSearch/Wrap/rdSynthonSpaceSearch.cpp

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright (C) David Cosgrove 2024.
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include <RDBoost/python.h>
+
+#include <GraphMol/ROMol.h>
+#include <GraphMol/SynthonSpaceSearch/SynthonSpace.h>
+
+namespace python = boost::python;
+
+namespace RDKit {
+
+python::list hitMolecules_helper(
+    const SynthonSpaceSearch::SubstructureResults &res) {
+  python::list pyres;
+  for (auto &r : res.getHitMolecules()) {
+    pyres.append(boost::shared_ptr<ROMol>(new ROMol(*r)));
+  }
+  return pyres;
+}
+
+struct SubstructureResults_wrapper {
+  static void wrap() {
+    std::string docString = "Used to return results of SynthonSpace searches.";
+    python::class_<SynthonSpaceSearch::SubstructureResults>(
+        "SubstructureResult", docString.c_str(), python::no_init)
+        .def("GetHitMolecules", hitMolecules_helper, python::args("self"),
+             "A function returning hits from the search")
+        .def_readonly(
+            "GetMaxNumResults",
+            &SynthonSpaceSearch::SubstructureResults::getMaxNumResults,
+            "The upper bound on number of results possible.  There"
+            " may be fewer than this in practice for several reasons"
+            " such as duplicate reagent sets being removed or the"
+            " final product not matching the query even though the"
+            " synthons suggested they would.");
+  }
+};
+
+SynthonSpaceSearch::SubstructureResults substructureSearch_helper(
+    SynthonSpaceSearch::SynthonSpace &self, const ROMol &query,
+    const python::object &py_params) {
+  SynthonSpaceSearch::SynthonSpaceSearchParams params;
+  if (!py_params.is_none()) {
+    params = python::extract<SynthonSpaceSearch::SynthonSpaceSearchParams>(
+        py_params);
+  }
+  auto results = self.substructureSearch(query, params);
+  return results;
+}
+
+void summariseHelper(SynthonSpaceSearch::SynthonSpace &self) {
+  self.summarise(std::cout);
+}
+
+BOOST_PYTHON_MODULE(rdSynthonSpaceSearch) {
+  python::scope().attr("__doc__") =
+      "Module containing implementation of SynthonSpace search of"
+      " Synthon-based chemical libraries such as Enamine REAL.";
```

Comment:
Please add a note that the API  here is experimental and may change. There's an example of this in the MIF PR:
https://github.com/rdkit/rdkit/blob/44e16fd17c6bd638c1856fb452c6d92151498003/Code/GraphMol/MolInteractionFields/Wrap/rdMIF.cpp#L30